syntax = "proto3";

package cust_reference;

import "google/protobuf/empty.proto";

option go_package = "github.com/milkyhoop/cust_reference/internal/delivery/pb/reference;reference";

// Indonesian Reference Resolution Service
service Cust_reference {
    // Resolve Indonesian pronouns to entities
    rpc ResolveReference(ReferenceRequest) returns (ReferenceResponse);
    
    // Health check
    rpc Health(google.protobuf.Empty) returns (HealthResponse);
}

// Reference resolution request
message ReferenceRequest {
    string session_id = 1;
    string tenant_id = 2;
    string reference_text = 3;  // "yang tadi", "yang paling murah", etc.
    string context_query = 4;   // Full user query for context
}

// Reference resolution response  
message ReferenceResponse {
    bool success = 1;
    string resolved_entity = 2;     // Resolved entity name
    string entity_type = 3;         // "product", "service", etc.
    string resolution_method = 4;   // "focus", "filter", "last_mentioned"
    repeated string candidates = 5;  // Alternative candidates
    string error_message = 6;
}

// Health check response
message HealthResponse {
    string status = 1;
    string service = 2;
    int64 timestamp = 3;
}
